src/main/java/duke/Task.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import java.time.LocalDate;
+
+public abstract class Task {
+    /**
+     * Description of the task.
+     */
+    private String taskName;
+    /**
+     * Information of task completion.
+     */
+    private boolean completed;
```

Comment:
Might be more appropriate here to name `completed` as `isCompleted` to be consistent with the coding standards defined for CS2103T though there are certainly cases where I would prefer `completed` to `isCompleted`